### PR TITLE
use top-domain module instead of hand rolled function

### PR DESCRIPTION
**Pull request by f2prateek**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -354,11 +354,11 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
     var writeKey = this.options.apiKey;
 
     // Exclude the current domain from the list of servers we're querying
-    var currentTld = getTld(window.location.hostname);
+    var currentTld = topDomain(window.location.hostname);
     var domains = [];
     for (var i=0; i<this.options.crossDomainIdServers.length; i++) {
       var domain = this.options.crossDomainIdServers[i];
-      if (getTld(domain) !== currentTld) {
```
Comment (tonyxiao) [inline lib/index.js]: @f2prateek this might not actually work.

Comment (tonyxiao) [reply lib/index.js]: the top domain function does the following
```
you can't set cookies on one of those tld suffixes

[11:45 AM]  
so it just tries the smallest suffix up until it finds one you can set cookies on
```

Comment (tonyxiao) [reply lib/index.js]: Wait actually it does work due to the fact that you will be able to get the top domain of the xid server that corresponds to current domain.. I wonder if we can add a test case for this at all though.

**Comment by tonyxiao** [comment]
Can we either add a comment explaining the extremely tricky behavior of `topDomain` or have some sorta test case for it?

File: lib/index.js [hunk 2]
```diff
@@ -354,11 +354,11 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
     var writeKey = this.options.apiKey;
 
     // Exclude the current domain from the list of servers we're querying
-    var currentTld = getTld(window.location.hostname);
+    var currentTld = topDomain(window.location.hostname);
```
Comment (tonyxiao) [inline lib/index.js]: we should probably add a requirement in our `package.json` rather than assume this function exists?

Comment (f2prateek) [reply lib/index.js]: We're already using it

See: 

1. https://github.com/segment-integrations/analytics.js-integration-segmentio/blob/791b9ddf45e2c5efb62697574e4077cc94d4b607/lib/index.js#L18 
2. https://github.com/segment-integrations/analytics.js-integration-segmentio/blob/791b9ddf45e2c5efb62697574e4077cc94d4b607/package.json#L33 
3. https://github.com/segment-integrations/analytics.js-integration-segmentio/blob/791b9ddf45e2c5efb62697574e4077cc94d4b607/lib/index.js#L301

File: lib/index.js (hunk 1 above)
Comment (f2prateek) [reply lib/index.js]: I couldn't think of a good way to test this but if you have a suggestion I can add a test.

**Comment by f2prateek** [comment]
@tonyxiao added a comment, let me know if you have strategies on testing.

**Comment by tonyxiao** [comment]
This is super tough to automatically test. Then again, since the worst case scenario is that we make an extra request that's not needed I don't feel we need to be as dogmatic about it. Happy to merge it assuming circle passes.

**Comment by f2prateek** [comment]
@ndhoule @tsholmes do you have any strategies to try to test this?

**Review by tonyxiao — APPROVED** [review]

**Pull request merged**